slimt/Annotation.hh

```diff
@@ -76,6 +76,21 @@ class Annotation {
     return Range{token_begin_[token_idx], token_begin_[token_idx + 1]};
   }
 
+  void update_words(const std::vector<Range> &words) {
+    if (words.empty()) {
+      return;
+    }
+
+    token_begin_.clear();
+    token_begin_.push_back(0);
+
+    for (const auto &word : words) {
+      token_begin_.push_back(word.begin);
+    }
+    token_begin_.push_back(words.back().end);
+    token_begin_.push_back(words.back().end);
```

Comment:
Are we sure the extra `end` is required? Can we have some comments explaining stuff?

Reply:
https://github.com/jerinphilip/slimt/pull/38/commits/5d49b1099513037f81add35bc9e81bf8a87e6690